docs/forms-app-environments.md

```diff
@@ -0,0 +1,194 @@
+  formsAppEnvironmentId,
+)
+```
+
+### Parameters
+
+| Parameter               | Required | Type     | Description                                               |
+| ----------------------- | -------- | -------- | --------------------------------------------------------- |
+| `formsAppEnvironmentId` | Yes      | `number` | The exact id of the forms app environment you wish to get |
+
+### Result (Resolved Promise)
+
+```json
+{
+  "id": 2,
+  "name": "Development",
+  "description": "This is the greatest environment ever made",
+  "organisationId": "59a17b708b1f8f8700ee1709",
+  "slug": "dev",
+  "createdAt": "2018-06-18T06:28:28.000Z",
+  "updatedAt": "2018-06-18T06:28:42.000Z"
+}
+```
+
+## `search()`
+
+### Example
+
+```javascript
+const options = {
+  limit: 1,
+  offset: 1,
+}
+const { formsAppEnvironment } = await formsAppEnvironments.search(options)
+```
+
+### Parameters
+
+| Parameter        | Required | Type     | Description                                                                           |
+| ---------------- | -------- | -------- | ------------------------------------------------------------------------------------- |
+| `options`        | No       | `Object` | Search options.                                                                       |
+| `options.limit`  | No       | `number` | Limit the number of results returned                                                  |
+| `options.offset` | No       | `number` | Skip a specific number of results, used in conjunction with `limit` to enforce paging |
+
+### Result (Resolved Promise)
+
+```json
+{
+  "meta": {
+    "limit": null,
+    "offset": null
+  },
+  "formsAppEnvironments": [
+    {
+      "id": 2,
+      "name": "Development",
+      "slug": "dev",
+      "description": "This is the greatest environment ever made",
+      "organisationId": "59a17b708b1f8f8700ee1709",
+      "createdAt": "2018-06-18T06:28:28.000Z",
+      "updatedAt": "2018-06-18T06:28:42.000Z"
+    },
+    ...
+  ]
+}
+```
+
+## `createFormsAppEnvironment()`
+
+### Example
+
+```javascript
+const formsAppEnvironment =
+  await formsAppEnvironments.createFormsAppEnvironment({
+    name: 'string',
+    description: 'string',
+    slug: 'string',
+    organisationId: '5a57ff94ff150a0010aa96ea',
+    cloneOptions: {
+      sourceFormsAppEnvironmentId: 1,
+      isCloningFormElementOptionsSets: true,
+      isCloningFormElementLookups: true,
+      isCloningFormSubmissionEvents: true,
+      isCloningFormServerValidation: true,
+      isCloningFormExternalIdGeneration: true,
+      isCloningFormPostSubmissionActions: true,
+    },
+  })
+```
+
+### Parameters
+
+| Parameter             | Required | Type                     | Description                                 |
+| --------------------- | -------- | ------------------------ | ------------------------------------------- |
+| `formsAppEnvironment` | Yes      | `NewFormsAppEnvironment` | The forms app environment object to create. |
```

Comment:
Do we need a definition for this type and `FormsAppEnvironment` in the docs? Could do it like some of the form ones have been done, where the def is in another .md file and we link to it.

Reply:
Would be better to reference the types repo rather than redefining them

Reply:
I agree. I think I'm going to investigate some tools to generate docs from typescript